src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
Better to send also 2nd parameter DocumentImpl and name it startProcessing(Txn, DocumentImpl) & endProcessing(Txn, DocumentImpl) because there are xupdate cases (element and attribute) that process only part of document (one attribute).